concourse/steps/scan_sources.py

```diff
@@ -61,23 +72,36 @@ def scan_component_with_whitesource(
         whitesource_cfg_name=whitesource_cfg_name,
     )
 
-    product_name, projects = whitesource.util.scan_sources(
-        whitesource_client=whitesource_client,
+    scan_artifacts = whitesource.util.scan_artifacts_from_component_descriptor(
+        filters=filters,
         component_descriptor=component_descriptor,
+    )
+
+    whitesource.util.scan_artifacts(
+        whitesource_client=whitesource_client,
         extra_whitesource_config=extra_whitesource_config,
         max_workers=max_workers,
-        filters=filters,
+        scan_artifacts=scan_artifacts,
+    )
+
+    product = whitesource.util.product_for_component_descriptor(
+        whitesource_client=whitesource_client,
+        component_descriptor=component_descriptor,
+    )
+
+    whitesource.util.insert_results(
```

Comment:
I think I did not yet reach the whitesource-changes.

generally speaking, I would have expected the util to upload to compliance-db w/ the compliance-db-related package 🤔